midterm-2023-11-07/code/5089-jovan-karajanov/script.ts

```diff
@@ -0,0 +1,117 @@
+interface Movie {
+    id: number;
+    title: string;
+    director: string;
+    year: number;
+    genre: string;
+    plot: string;
+    cast: string[];
+    oscars: number;
+  }
+  
+  document.addEventListener("DOMContentLoaded", siteCode);
+  type MovieSorter = (first: Movie, second: Movie) => number;
+  let movies: Movie[] = [];
+  
+  async function siteCode() {
+    
+      const data = await loadData();
+      movies = data;
+      displayMovies(movies);
+  
+   
+    const idSort = document.getElementById("sort-id")!;
+    idSort.addEventListener("click", sortById);
+
+    const titleSort = document.getElementById("sort-title")!;
+    titleSort.addEventListener("click", sortByTitle);
+
+}
+
+  
+  async function loadData() {
+    const dataUri = "https://raw.githubusercontent.com/sweko/uacs-internet-programming-exams/main/midterm-2023-11-07/data/movies.json";
+    const response = await fetch(dataUri);
+  
+    if (!response.ok) {
+      throw new Error("The data is not available.");
+    }
+  
+    const data = await response.json();
+    return data;
+  }
+  
+  const idSort: MovieSorter = (first, second) => first.id - second.id;
+  const titleSort: MovieSorter = (first, second) => first.title.localeCompare(second.title);
+  
+  const sortById = () => {
+    const sortedMovies =movies.toSorted(idSort);
+    displayMovies(sortedMovies);
+  }
+
+  const sortByTitle = () => {
+    const sortedMovies = movies.toSorted(titleSort);
+  }
+
+  function displayMovies(movies: Movie[]) {
+    const container = document.getElementById("movie-container")!;
+    container.innerHTML = "";
+  
+    for (const movie of movies) {
+      const movieTable = generateMovieTable(movie);
+      container.appendChild(movieTable);
+    }
+  }
+
+/// Filters
+`const titleFilterInput = document.getElementById("title-filter") as HTMLInputElement;
```

Comment:
This code (written inside a string because of reasons 😄 ) will successfully get the filter values from the page +10